lxc/storage_volume.go

```diff
@@ -1111,13 +1112,29 @@ func (c *cmdStorageVolumeEdit) run(cmd *cobra.Command, args []string) error {
 		}
 
 		if isSnapshot {
-			newdata := api.StorageVolumeSnapshotPut{}
-			err = yaml.Unmarshal(contents, &newdata)
+			// Get the current data for validation
+			origdata, etag, err := client.GetStoragePoolVolumeSnapshot(resource.name, volType, fields[0], fields[1])
 			if err != nil {
 				return err
 			}
 
-			op, err := client.UpdateStoragePoolVolumeSnapshot(resource.name, volType, fields[0], fields[1], newdata, "")
+			// Unmarshal stdin to newdata
+			newdata := api.StorageVolumeSnapshot{}
+			err = yaml.Unmarshal(contents, &newdata)
+			if err != nil {
+				return fmt.Errorf(i18n.G("Failed to parse edited content: %w"), err)
+			}
+
+			// Create a copy of the original snapshot with only writable properties changed
+			expectedVol := *origdata
+			expectedVol.SetWritable(newdata.Writable())
+
+			// Validate that only writable properties were modified
+			if !reflect.DeepEqual(newdata, expectedVol) {
+				return fmt.Errorf("%s", i18n.G("Only 'description' and 'expires_at' fields can be modified for snapshot volumes"))
```

Comment:
Please can you use double quoting rather than single quoting.
Also lets avoid the use of `fmt.Errorf` when not using placeholders.
Instead use `errors.New()`.